src/Serilog.Sinks.Amazon.Kinesis/Common/HttpLogShipperBase.cs

```diff
@@ -268,40 +237,36 @@ static bool IsUnlockedAtLength(string file, long maxLen)
             catch (IOException ex)
             {
                 var errorCode = Marshal.GetHRForException(ex) & ((1 << 16) - 1);
-
-                if (errorCode == 32)
+                if (errorCode != 32 && errorCode != 33)
```

Comment:
looking at the seq sink, seems to make sense.

```cs
        const long ERROR_SHARING_VIOLATION = 0x20;
        const long ERROR_LOCK_VIOLATION = 0x21;

            catch (IOException ex)
            {
                long win32ErrorCode = ex.HResult & 0xFFFF;
                if (win32ErrorCode != ERROR_SHARING_VIOLATION && win32ErrorCode != ERROR_LOCK_VIOLATION ) {
                    Logger.TraceException("Unexpected I/O exception while testing locked status of {0}", ex, file);
                }
            }
            catch (Exception ex) {
                Logger.TraceException("Unexpected exception while testing locked status of {0}", ex, file);
            }
```

so it will swallow any file sharing/lock exceptions and only propagate other exceptions...

I wonder if there are typically a tonne of such exceptions being swallowed and ignored in the normal course of events...?